.travis.yml

```diff
@@ -0,0 +1,864 @@
+        - git clone https://github.com/iovisor/bcc
+        - cd bcc
+        - git checkout remotes/origin/tag_v0.4.0
+        - mkdir -p _build
+        - cd _build
+        - cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+        - make
+        - sudo make install
+        - cd ../..
+        - cargo build --features $BCC
+        - cargo test --features $BCC
+        - cargo build --release --features $BCC
+        - cargo test --release --features $BCC
+        - sudo target/release/runqlat --interval 1 --windows 5
+    - name: "trusty / llvm 5 / bcc 0.5.0"
+      os: linux
+      dist: trusty
+      rust: stable
+      env: BCC=bcc_v0_5_0 RUST_BACKTRACE=1
+      script:
+        - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+        - echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-5.0 main" | sudo tee -a /etc/apt/sources.list
+        - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/ppa
+        - sudo apt-get update
+        - sudo apt-get install linux-headers-`uname -r`
+        - sudo apt-get --yes install bison build-essential cmake3 flex git libclang-common-5.0-dev libelf-dev libllvm5.0 libz-dev lldb-5.0 llvm-5.0 llvm-5.0-dev llvm-5.0-runtime
+        - git clone https://github.com/iovisor/bcc
+        - cd bcc
+        - git checkout remotes/origin/tag_v0.5.0
+        - mkdir -p _build
+        - cd _build
+        - cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+        - make
+        - sudo make install
+        - cd ../..
+        - cargo build --features $BCC
+        - cargo test --features $BCC
+        - cargo build --release --features $BCC
+        - cargo test --release --features $BCC
+        - sudo target/release/runqlat --interval 1 --windows 5
+    - name: "trusty / llvm 5 / bcc 0.6.0"
+      os: linux
+      dist: trusty
+      rust: stable
+      env: BCC=bcc_v0_6_0 RUST_BACKTRACE=1
+      script:
+        - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+        - echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-5.0 main" | sudo tee -a /etc/apt/sources.list
+        - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/ppa
+        - sudo apt-get update
+        - sudo apt-get install linux-headers-`uname -r`
+        - sudo apt-get --yes install bison build-essential cmake3 flex git libclang-common-5.0-dev libelf-dev libllvm5.0 libz-dev lldb-5.0 llvm-5.0 llvm-5.0-dev llvm-5.0-runtime
+        - git clone https://github.com/iovisor/bcc
+        - cd bcc
+        - git checkout remotes/origin/tag_v0.6.0
+        - mkdir -p _build
+        - cd _build
+        - cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+        - make
+        - sudo make install
+        - cd ../..
+        - cargo build --features $BCC
+        - cargo test --features $BCC
+        - cargo build --release --features $BCC
+        - cargo test --release --features $BCC
+        - sudo target/release/runqlat --interval 1 --windows 5
```

Comment:
You need to add your binary here as well. Will require wrapping with `timeout` command. See github.com/twitter/rezolus for example. 

I'm unsure why the diff for this file is so large, since we do have the CI config in master now.

Reply:
@brayniac 
With the `timeout` command, is there a time limit we wish to set for this.  I've placed a 1 minute limit for this to run for now.  I wonder how this will impact the travis CI free build time budget per month.

Reply:
@afbase - I am only aware of limits per-job. I believe it's important we're only releasing tested code. From the docs, I can only see limits per-job, where the build will not timeout unless any job within it does. (https://docs.travis-ci.com/user/customizing-the-build#build-timeouts)

If we look for the new tool, we've caught an issue on bcc 0.4.0: https://travis-ci.org/rust-bpf/bpf-tools/jobs/656075826#L1857

It's possible we'll want to restrict our compatibility to some subset of recent bcc versions, but illustrates the importance of having a ci matrix and running our tools through it.

Reply:
> illustrates the importance of having a ci matrix and running our tools through it.

1. I really like the build matrix you have going on here. 

>we'll want to restrict our compatibility to some subset of recent bcc versions

would we have to restrict the build for tcpconnect to exclude `bcc_v0_4_0`?  Could you elaborate a bit more on this?  

>If we look for the new tool, we've caught an issue on bcc 0.4.0: https://travis-ci.org/rust-bpf/bpf-tools/jobs/656075826#L1857

1. TIL travis gives line numbers for build STDOUT!  this is a super handy feature.

Reply:
I was trying to say that we might want to drop anything older than maybe bcc 0.10.0 for the whole repo. If we can make the tcpconnect tool work on bcc 0.4.0, that would be preferred until we decide on a particular oldest bcc version that we want to support the tools on.

Reply:
I'm still playing with `bpf.c` to see if I can get a variant to work with bcc 0.4.0. Giving myself a little more time.

Reply:
hi @brayniac,
> was trying to say that we might want to drop anything older than maybe bcc 0.10.0 for the whole repo. If we can make the tcpconnect tool work on bcc 0.4.0

I'm leaning towards dropping support for anything older than 0.10.0.  I've spent a considerable amount of time on trying to get bcc 0.4.0 working but no such luck and I'd like to spend time on another project for ~1 month (I'll come back to bpf-tools and add more thereafter).

The code seems to work as low as bcc 0.5.0 for the time being.  

My thoughts are as follows:
1. remove tests in `travis.yml` for any `bcc 0.x.0` for some version less than or equal to `x`. 
1. version control bcc in the Cargo.toml dependencies. 

Let me know what you think.

Reply:
I think it's fair to drop anything pre 0.10.0. Should be able to simply remove prior versions from travis config and from the Cargo.toml features.